go/proxyd/backend.go

```diff
@@ -170,6 +171,7 @@ func NewBackend(
 	rpcURL string,
 	wsURL string,
 	rateLimiter RateLimiter,
+	rpcSemaphore *semaphore.Weighted,
```

Comment:
IMO it'd be cleaner if we passed in the max concurrent RPCs rather than the semaphore itself. This way we can abstract away how the concurrency is limited.

Reply:
That wouldn't work because each backend as there'll be effective be `max_concurrent_rpcs * num_backends` max concurrent rpcs. So we need a shared concurrency limiter that's used across all backends. The semaphore itself is simple enough for its purpose that abstracting it wouldn't provide any benefit imo.

Reply:
Aha, thanks for clearing that up.